File: GvgAmpClient/test/gvgampclient/test_gvgampclient.cpp
/** \example test_gvgampclient.cpp
* \brief A short example of how to use the GvgAmpClient class in C++.
*/


#include <stdio.h>
#include <stdlib.h>
#include <iostream>
#include <gvgampclient/gvgampclient.h>

int main(int argc, char *argv[])
{
   using namespace gvgampclient;
   std::vector<std::string> folders;
   std::vector<std::string> files;
   std::string cwd;

   if(argc<2) {
       printf("Please specify host as the first argument.\n");
       exit(-1);
   }
   
   try {
       GvgAmpClient gvg_client(argv[1]);
       printf("Looks like it's connected\n");
  
       gvg_client.ListFirstFolder(folders);
       gvg_client.ListNextFolder(folders);       
       std::cout << "Listing folders:\n";
       for( std::vector<std::string>::const_iterator i = folders.begin(); 
                                                     i != folders.end(); ++i)
           std::cout << "   " << *i << '\n';   
       std::cout << "Done.\n"; 

       std::cout << "Listing files:\n";
       gvg_client.ListFirstID(files,1);
       gvg_client.ListNextID(files,10,1);
       for( std::vector<std::string>::const_iterator i = files.begin(); 
                                                     i != files.end(); ++i)
           std::cout << "   " << *i << '\n';   
       std::cout << "Done.\n"; 
  
       gvg_client.GetWorkingFolder(cwd);
       std::cout << "Current working folder:" << cwd << "\n";

  }
   catch(SocketException &e) {
       printf("Received an exception:%s\n",e.description().c_str());
   }
}


   

File: GvgAmpClient/test/gvgampclient/test_gvgampclient.php
#!/usr/bin/php -f
<?php
/** \example test_gvgampclient.php
* \brief A short example of how to use the GvgAmpClient class in PHP.
*/


  include("pGvgAmpClient.php");
  $folders=array();
  $clips=array();
  $result=0;
  $cwd="";   //The working directory
  $afd=0;    //The AFD value
  $ws=0;     //The widescreen value
  try {  
      $client = new GvgAmpClient("192.168.10.25");
     
      echo "Listing folders...\n";
      $result=$client->ListFolders($folders);
      foreach ($folders as $folder) {
         echo "   ".$folder."\n";
      }
      echo "Total folders:".$result."\n";

      echo "Current working folder: ";

      $client->GetWorkingFolder($cwd);
      echo $cwd."\n";

      echo "Set working folder to TESTBE: ";
      if ($client->SetWorkingFolder("TESTBE")==0) 
         echo "SUCCESS\n";
      else {
         echo "FAILED. Exiting...\n";
         goto DISCONNECT;
      }

      echo "Current working folder: ";
      $client->GetWorkingFolder($cwd);
      echo $cwd."\n";

      echo "Set working folder to EDIUS: ";
      if ($client->SetWorkingFolder("EDIUS")==0) 
         echo "SUCCESS\n";
      else {
         echo "FAILED. Exiting...\n";
         goto DISCONNECT;
      }

      echo "Current working folder: ";
      $client->GetWorkingFolder($cwd);
      echo $cwd."\n";

      echo "Listing clips...\n";
      echo "Clipname                               AFD Flag     Widescreen Flag";
      echo "-------------------------------------------------------------------";
      $client->ListIDs($clips,1);
      foreach($clips as $clip) {
          $client->GetAFDSetting($clip,$afd,$ws);
          echo $clip."                         ".$afd."      ".$ws."\n";
      }
  } catch (SocketException $e) {
      echo "Exception:".$e.describe()."\n";
  }      
  DISCONNECT:
  unset($client);

?>
